src/Stratis.Bitcoin/Base/AsyncWork/IAsyncDelegateDequeuer.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace Stratis.Bitcoin.Base.AsyncWork
+{
+    public interface IAsyncDelegateDequeuer : IDisposable { }
```

Comment:
This type is redundant

Reply:
actually it's needed as key for `asyncDelegateWorkers` dictionary

Reply:
Ah right, why dont we create a shared interface for that? and merge both dictionaries? is there a benefit in the separation?

Then we would call it `AsyncProviderItem` ?
Just a thought I am not picky here.

Reply:
refer to https://github.com/stratisproject/StratisBitcoinFullNode/pull/3455/files/47f00393c4d0a78a66f52454a18bbe4091e078fe#r271742636